### PR TITLE
Updated DCGAN lab

### PR DESCRIPTION
**Pull request by takumiohym**
- Moved DCGAN lab from supplemental dir to image_models dir
- Minor updates including cell order change

## Test environment
- Vertex AI Workbench TensorFlow 2.8
- Attach 1 GPU (T4) to speed up training

### CODE REVIEW

**Comment by BenoitDherin** [comment]
@takumiohym It's taking a really long time to complete. Should it be run on GPU. Please indicate the appropriate setup to test this lab.

**Comment by BenoitDherin** [comment]
@takumiohym Please submit different PRs for solutions and labs moving forward.

File: notebooks/image_models/solutions/deepconv_gan.ipynb [hunk 1]
```diff
@@ -50,33 +50,6 @@
     "## Import TensorFlow and other libraries"
```
Comment (BenoitDherin) [inline notebooks/image_models/solutions/deepconv_gan.ipynb:50]: <p>Move this install into the top level <code>requirements.txt</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/221/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (BenoitDherin) [inline notebooks/image_models/solutions/deepconv_gan.ipynb:50]: <p>Remove the Ipython import which is no longer used</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/221/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (BenoitDherin) [inline notebooks/image_models/solutions/deepconv_gan.ipynb:50]: <p>Explain what <code>from_logits=True</code> means and why we need to use that (i.e. the discriminator outputs logits here and not probabilities)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/221/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='30'/>

Comment (BenoitDherin) [inline notebooks/image_models/solutions/deepconv_gan.ipynb:50]: <p><code>noise_dim</code> and <code>num_examples_to_generate</code> in upper caps  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/221/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='42'/>

**Review by BenoitDherin — APPROVED** [review]
LGTM. Minor changes. Also maybe add a WARNINGS file in the image folder at the same level as labs and solutions with the proper setup for this lab to be run and the time it takes to run it.

File: notebooks/image_models/solutions/deepconv_gan.ipynb (hunk 1 above)
Comment (takumiohym) [reply notebooks/image_models/solutions/deepconv_gan.ipynb:50]: <p>Make sense. I added. Thanks!</p>

Comment (takumiohym) [reply notebooks/image_models/solutions/deepconv_gan.ipynb:50]: <p>Agreed. Thanks!</p>

Comment (takumiohym) [reply notebooks/image_models/solutions/deepconv_gan.ipynb:50]: <p>It seems imageio is preinstalled and this line is not needed.</p><p>I will just remove it so far. Thanks!</p>

Comment (takumiohym) [reply notebooks/image_models/solutions/deepconv_gan.ipynb:50]: <p>Actually <code>IPython.display</code>  module is used in <code>train</code> function below. </p>

Comment (takumiohym) [reply notebooks/image_models/solutions/deepconv_gan.ipynb:50]: <p>Actually <code>IPython.display</code> is used in <code>train</code> function. Do you think we can keep it?  </p>

**Comment by takumiohym** [comment]
> @takumiohym It's taking a really long time to complete. Should it be run on GPU. Please indicate the appropriate setup to test this lab. Maybe add a WARNINGS file in the directory and list the particular JupyterLab setup to run this lab smoothly.

Right. I added a testing environment at the top, and will do the same to the other PRs.
Thanks!

**Comment by takumiohym** [comment]
> LGTM. Minor changes. Also maybe add a WARNINGS file in the image folder at the same level as labs and solutions with the proper setup for this lab to be run and the time it takes to run it.

Thank you!
In terms of `imageio` module, it seems it is pre-installed in Vertex AI Wokrbench. So this time I simply removed the line.

**Review by kylesteckler — APPROVED** [review]
